R/process_sc_all_care_home.R

```diff
@@ -28,53 +29,63 @@ process_sc_all_care_home <- function(
     spd_path = read_file(get_spd_path()),
     write_to_disk = TRUE) {
   ## Data Cleaning-----------------------------------------------------
+
+  # to do:
+  # compare old to new
+  # PR
+
   ch_clean <- data %>%
     dplyr::mutate(
-      record_date = end_fy_quarter(.data[["period"]]),
-      qtr_start = start_fy_quarter(.data[["period"]]),
-      # Set missing admission date to start of the submitted quarter
+      # Set missing admission date to start of the submitted quarter (n = 2)
       ch_admission_date = dplyr::if_else(
         is.na(.data[["ch_admission_date"]]),
-        .data[["qtr_start"]],
+        .data[["period_start_date"]],
         .data[["ch_admission_date"]]
       ),
-      # TODO check if we should set the dis date to the end of the period?
-      # If the dis date is before admission, remove the dis date
+      # If the dis date is before admission, remove the dis date (n = 5)
       ch_discharge_date = dplyr::if_else(
         .data[["ch_admission_date"]] > .data[["ch_discharge_date"]],
         lubridate::NA_Date_,
         .data[["ch_discharge_date"]]
       )
     ) %>%
-    dplyr::left_join(sc_demog_lookup,
+    dplyr::full_join(sc_demog_lookup, # this is the correct join.
       by = c("sending_location", "social_care_id")
     ) %>%
-    replace_sc_id_with_latest()
+    replace_sc_id_with_latest() %>%
+    dplyr::select(-latest_flag, -latest_sc_id)
 
+
+  # cleaning and matching care home names -  Zihao looking into care home names
   name_postcode_clean <- fill_ch_names(
     ch_data = ch_clean,
     ch_name_lookup_path = ch_name_lookup_path,
     spd_path = spd_path
   )
 
+
   fixed_ch_provider <- name_postcode_clean %>%
+    dplyr::select(-ch_name_validated, -open_interval, -latest_close_date, -ch_name_old, -ch_postcode_old) %>%
     dplyr::mutate(
-      ch_provider = dplyr::if_else(is.na(.data[["ch_provider"]]), 6L, .data[["ch_provider"]])
+      ch_provider = dplyr::if_else(is.na(.data[["ch_provider"]]), 6L, .data[["ch_provider"]]) # (n = 2)
     ) %>%
     # sort data
     dplyr::arrange(
-      "sending_location",
-      "social_care_id",
-      "ch_admission_date",
-      "period"
+      .data[["sending_location"]],
+      .data[["social_care_id"]],
+      .data[["period"]],
+      .data[["ch_admission_date"]]
```

Comment:
Agree this new sort is working correctly and now the dates are in the right order for later.